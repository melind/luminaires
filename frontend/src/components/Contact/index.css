.contact {
    border: yellow solid 2px;
   
}

Form {
    border: yellow solid 2px;
}