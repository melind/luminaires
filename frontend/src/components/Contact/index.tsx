import React, { useState } from 'react';
import { Form, Input, Button } from 'antd';
import  displayError  from '../../lib/validation';
import userAPI from '../../services/userAPI';

import './index.css';


const Contact = ({name, mail, subject, message, onSubmit, error}) => {
console.log("states come from reducer:", name, mail, subject, message,error);
userAPI.contact();

    const layout = {
      labelCol: { span: 8 },
      wrapperCol: { span: 8 },
    };
  



    const [formState, setFormState] = useState({name, mail, subject, message});
            const handleChange = (e) => {
                const name: string = e.target.name;
                const value: string = e.target.value;
                console.log("name : ",name, "value :", value);
                setFormState({...formState, [name]: value}); // name_input : input_value   

            }
    const result = displayError(formState);
        console.log("result: ", result, "form: ", formState);
    const handleSubmit = (e) => {
        e.preventDefault();
        
        console.log("erreur: ",error);
        if (!result[0]) { 
        onSubmit(formState);
        
        console.log("ok");
        }
   
    }
  
    return (

 
        <div className="contact" > 
            Contact
             <Form {...layout} name="nest-messages"  onSubmit={handleSubmit} method="post" action="mailto:melinda93@hotmail.fr" >
                 <Form.Item label="Nom" >
                   <Input name="name" onChange={handleChange}/>
                 </Form.Item>
                 <Form.Item label="Email"  >
                   <Input name="mail" onChange={handleChange}/>
                 </Form.Item>
                 <Form.Item  label="Sujet">
                   <Input name="subject" onChange={handleChange} />
                 </Form.Item>
                 <Form.Item label="Message">
                   <Input.TextArea  name="message" onChange={handleChange} rows={4}/>
                 </Form.Item>
                 <Form.Item wrapperCol={{ ...layout.wrapperCol, offset: 8 }}>
                   <Button type="primary" htmlType="submit"> 
                     Submit 
                   </Button>
                 </Form.Item> 
            </Form>
        </div>
    )
}

export default Contact;