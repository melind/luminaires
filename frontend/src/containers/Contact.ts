import { connect } from 'react-redux';
import Contact from '../components/Contact';
import {send}  from '../store/reducer/contact-reducer';

const mapStateToProps = (state) => ({
    name: state.send.name,
    mail: state.send.mail,
    subject: state.send.subject,
    message: state.send.message,
    error: state.send.error
});
const r= "e";
const mapDispatchToProps = (dispatch) => ({
    onSubmit: (formState) => {
        dispatch(send(formState)); 
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Contact);