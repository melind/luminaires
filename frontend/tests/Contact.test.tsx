import React from 'react';
import renderer from 'react-test-renderer'
import { shallow, mount} from 'enzyme';
import Contact from '../src/components/Contact';

/* test de rendu*/
describe('Home render', () => {
it('Contact snapshot', () => {
  const tree = renderer
    .create(<Contact />).toJSON();//use render() of component
  expect(tree).toMatchSnapshot();
});
/* test de fonctionnalitÃ©*/
 /* unit test*/

 it('contact render one component', () => {
     const wrapper = shallow(<Contact optInValue={true} authenticated={true}/>);
     expect(wrapper.length).toEqual(1);  // to have one Component
  }); 


});