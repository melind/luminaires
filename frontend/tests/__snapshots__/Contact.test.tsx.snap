// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home render Contact snapshot 1`] = `
<div
  className="contact"
>
  Contact
</div>
`;
